File: Napakalaki Java/NapakalakiGame/DeathBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author Germán
 */
public class DeathBadConsequence extends NumericBadConsequence{
    private boolean death; //es o no mal rollo de tipo muerte
    
    public DeathBadConsequence(String Text, int Levels) {
        super (Text,Levels,5,5);
        death = true;
    }
    
    public boolean getDeath () {
        return death;
    }
    
    @Override
    public String toString() {
        return "Este mal rollo te mata";
    }
}

File: Napakalaki Ruby/DeathBadConsequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame

class DeathBadConsequence < NumericBadConsequence
  
  def initialize(texto, muerte)
    super(texto,0,0,0)
    @death = muerte
  end
  
  def getDeath
    
    return @death
  end
  
  def to_s
    cadena = "Este mal rollo te mata"
    return cadena
  end
  
  def adjustToFitTreasureList(v, h)
      malrollo = NumericBadConsequence.new(@text,0,v.size,h.size)
      return malrollo
  end
  
end

end
